test/smoke/verifymanager_test.go

```diff
@@ -0,0 +1,177 @@
+package smoke
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github.com/target/goalert/permission"
+	"github.com/target/goalert/test/smoke/harness"
+)
+
+// TestUserVerificationCleanup tests that verification codes are cleaned up
+// when they are expired. Also tests that contact methods that are not verified
+// are cleaned up with the verification code.
+func TestUserVerificationCleanup(t *testing.T) {
```

Comment:
This test is a little hard to follow. Also, it's several separate tests. `t.Run` is generally intended for table tests (i.e., from a `for`-loop), which is different from how test suites work in the js world.

I'd suggest just one test to validate this feature:
- two existing contact methods in the initial SQL, A, and B, where B is disabled
- create two with GraphQL, C and D; verify C
- assert that only D is deleted after fast-forwarding

That will validate that existing contact methods aren't deleted and that new ones will expire & get cleaned up if not verified.

Reply:
I've refactored the test.